### DISCUSSION TITLE
buy it now (on addtocart and list)

### DISCUSSION THREAD

**Discussion by f1-outsourcing** [OpenMage/magento-lts#4168] [Questions & Answers]
this is a nice solution for adding a button to buy a product quickly, going directly to checkout (although it does not clear your cart first)

https://magento.stackexchange.com/questions/123072/magento-1-9-how-to-add-buy-now-button-in-product-page

However this is a solution for the product page. I would like to have this button also on the list.phtml.

on the addtocart.phtml we have this 
`onclick="productBuyNowForm.submit(this)"`

on the list.phtml we have this
`$this->getAddToCartUrlCustom($_product, [], false); `
`onclick="customFormSubmit('<?php echo $_cartUrl ?>', '<?php echo $_params ?>', 'post')"`

this carUrl is something like this:
..../checkout/cart/add/uenc/aHR0cDovLzE5Mi4xNjguMTEuMTkvc2hvcC9lbl9nYi9rZW1wby5odG1s/product/13/

I was wondering what the method/url is for adding the product to the cart and going to checkout from the list overview. If that is even possible on this page.

**Comment by JesperHerrloff** [comment]
You could use an observer that redirects to checkout after product is added to Cart.

**Comment by kiatng** [comment]
Ref  `getAddToCartUrlCustom()`, you can set the attribute `add_to_cart_url_custom` in the product page.  You can define a custom controller action to add the product to cart and redirect to the cart or checkout page.

![image](https://github.com/user-attachments/assets/e3d8e4da-4fca-499a-b2a2-4ef0db7cd1c8)

But you need to do it for every product, which is a hassle, even with mass action. A solution is to observe these events to set the `add_to_cart_url_custom`:

- `catalog_product_collection_load_after` (on list page)
- `catalog_product_load_after` (on product page)
